chann.go

```diff
@@ -208,14 +210,13 @@ func (ch *Chann[T]) unboundedTerminate() {
 	}
 	for len(ch.q) > 0 {
 		select {
+		// Note if receiver doesn't consume all data that has been sent to input
+		// channel, the `unboundedProcessing` goroutine will leak forever.
+		// Ref: https://github.com/golang-design/chann/issues/3
 		case ch.out <- ch.q[0]:
-		// The default branch exists because we need guarantee
-		// the loop can terminate. If there is a receiver, the
-		// first case will ways be selected. See #3.
-		default:
+			ch.q[0] = nilT // de-reference earlier to help GC
+			ch.q = ch.q[1:]
```

Comment:
This reads like [a previous implementation](https://github.com/golang-design/chann/blob/619cda11e36af41418107ff5599a97ab71b9a418/chann.go#L173-L175) that was reported to be problematic, as discussed in #3. Could you help me to understand how your changes can improve the overall situation?

Reply:
Yep, in this PR, it behaves as follows
When an unbounded channel is closed
- If remaining data in input channel or buffer queue is larger than size of output channel, and no receiver will consume these data, the `unboundedProcessing` goroutine will block at https://github.com/golang-design/chann/pull/5/files#diff-95e80cac22cfd39c19faf1211d443965c49871890d6f92ee5d93fa719fc5f5daR214. **This situation leaks goroutine as well as memory of the chann, but user can consume all data to release these resources.**. At the same time there is no cpu spin issue.
- If remaning data can be sent to output channel, there is no goroutine leak issue.